.github/workflows/tests_archdetect_nvidia_gpu.yml

```diff
@@ -0,0 +1,124 @@
+# documentation: https://help.github.com/en/articles/workflow-syntax-for-github-actions
+name: Tests for accelerator detection (NVIDIA GPU)
+on:
+  push:
+  pull_request:
+permissions:
+  contents: read # to fetch code (actions/checkout)
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        fake_nvidia_smi_script:
+          - none # no nvidia-smi command
+          - no_devices # nvidia-smi command works, but no GPUs available
+          - 1xa100 # cc80, supported with (atleast) zen2 CPU
+          - 2xa100 # cc80, supported with (atleast) zen2 CPU
+          - 4xa100 # cc80, supported with (atleast) zen2 CPU
+          - cc01 # non-existing GPU
+      fail-fast: false
+    steps:
+    - name: checkout
+      uses: actions/checkout@d632683dd7b4114ad314bca15554477dd762a938 # v4.2.0
+
+    # we deliberately do not use the eessi/github-action-eessi action,
+    # because we want to control when the EESSI environment is initialized
```

Comment:
The required level of control is actually quite trivial if we use the module initialisation (`module unload EESSI`). We can centrally improve our action to support this scenario rather than suggest it is not (or cannot be) flexible...open an issue?

Reply:
I'm not saying it's impossible, just that there's no control currently for only mounting and not setting up the environment.

Issue opened @ https://github.com/EESSI/github-action-eessi/issues/28